src/lib/modules/deployment/contract_deployer.js

```diff
@@ -193,11 +193,18 @@ class ContractDeployer {
     contract.deployedAddress = trackedContract.address;
     self.events.emit("deploy:contract:deployed", contract);
 
-    // TODO: can be moved into a afterDeploy event
-    // just need to figure out the gasLimit coupling issue
-    self.events.request('code-generator:contract:vanilla', contract, contract._gasLimit || false, (contractCode) => {
-      self.events.request('runcode:eval', contractCode, () => {}, true);
-      return callback();
+    self.events.request('code-generator:contract:custom', contract, (customContractCode) => {
+      if (!customContractCode) {
```

Comment:
Notice that we first check if there's a custom contract code generator registered that returns dedicated code. We fallback to the vanilla contract creation in case non had been registered.

Reply:
Also notice that this gets executed for *all* contracts, including the ones that Embark comes with (e.g. `ENSRegistry` etc).

We might want to check for built-in vs owned contracts and only do this for the ones owned by the developer.

Thoughts?